src/app/settings/views/Scripts/ScriptsList/ScriptsList.test.tsx

```diff
@@ -212,14 +235,24 @@ describe("ScriptsList", () => {
         </MemoryRouter>
       </Provider>
     );
+    let row = screen.getByRole("row", { name: "commissioning-script" });
+    expect(row).not.toHaveClass("is-active");
     // Click on the delete button:
-    wrapper.find("TableRow").at(1).find("Button").at(1).simulate("click");
+    await userEvent.click(
+      within(within(row).getByLabelText(ScriptsListLabels.Actions)).getByRole(
+        "button",
+        { name: "Delete" }
+      )
+    );
     // Click on the delete confirm button
-    wrapper
-      .find("TableRow")
-      .at(1)
-      .find("ActionButton[data-testid='action-confirm']")
-      .simulate("click");
+    await userEvent.click(
+      within(
+        within(row).getByRole("gridcell", {
+          name: `Are you sure you want to delete Script "commissioning-script"? This action is permanent and can not be undone. Cancel Delete`,
```

Comment:
Could this target the row or the confirmation with an aria label? It would make this test less fragile.

Reply:
Yep, I can do that. For some reason I thought that wasn't possible, felt like I had a memory of it not being possible in similar tests, but I've found an example in RepositoriesList.test

https://github.com/canonical/maas-ui/blob/main/src/app/settings/views/Repositories/RepositoriesList/RepositoriesList.test.tsx#L117-L125